### ISSUE TITLE
residue extension missing from PHAF export

### ISSUE THREAD

**Issue by ValWood** [pombase/pombase-chado#814]
When "residue" is used as an extension:

cdr2 | cdr2Δ | FYPO:0002677 | abnormal protein phosphorylation | Gel electrophoresis evidence | residue K21

it does not seem to be exported into the pHAF file (I was looking for a syntax example but I don't find anything gripping the extension column for "residue")

**Comment by kimrutherford** [comment]
Should the residue be an extension or a new column?  If it was a column of its own it would be easier to parse and work with.

**Comment by ValWood** [comment]
In the HTP spreadsheet I made recently  (not yet checked in), I captured residue as an annotation extension. Then I noticed I could not see any in already  the PHAF download even though I know I have added them in Canto.

So if  the HTP file I just created is the only one PHAF file with residues , I guess a new column could be added, but it is used for so few phenotype terms it seems overkill  to add a new column. 

@mah11 what do you think?

**Comment by mah11** [comment]
Hmm. On one hand, we treat residues as extensions for FYPO annotations in all other contexts ... but on the other hand, the PHAFs already split severity and penetrance out from the other extensions (I assume that's for the same sort of easier parsing as here).

I suppose there's a sense of overkill in either adding a column to all PHAFs, or to doing more complicated extension parsing, for the residue extensions on FYPO annotations, simply because (I think) there aren't huge numbers of 'em. But one or the other has to be done, and adjusting the PHAFs -- and the documentation -- is a one-off.

tl;dr no strong preference because everything seems to cancel out!

**Comment by kimrutherford** [comment]
It's about the same amount of work for me to implement either plan.

**Comment by ValWood** [comment]
This file has now been committed, it currently has the residues as extensions
PMID_32062975_phaf.tsv

**Comment by mah11** [comment]
PMID_21561865_phaf.tsv also has residue extensions

**Comment by mah11** [comment]
I just got round to checking [PMID:21561865](https://www.pombase.org/reference/PMID:21561865) ... and the residue extensions are showing up just fine.

The actual problem is that the syntax was wrong - it should be `residue(S354)`, not `residue=S354`. Now that I've I corrected that, the residues will probably appear with the next update (reopen if they don't).

**Comment by kimrutherford** [comment]
> When "residue" is used as an extension:
> cdr2 | cdr2Δ | FYPO:0002677 | abnormal protein phosphorylation | Gel electrophoresis evidence | residue K21

Is this example from a gene page or from a file?  I can't see it on the cdr2 page.

**Comment by mah11** [comment]
>> cdr2 | cdr2Δ | FYPO:0002677 | abnormal protein phosphorylation | Gel electrophoresis evidence | residue K21
> 
> Is this example from a gene page or from a file? I can't see it on the cdr2 page.

Is it even a real example? If so, I suspect a copy/past mishap ... the "affected protein" isn't shown, and can't be cdr2 itself (its residue 21 is P). There's also no residue K21 anywhere on the [FYPO:0002677 term page](https://www.pombase.org/term/FYPO:0002677), but several other residue extensions do show up.

**Comment by ValWood** [comment]
I   don't think it is a real example.
IIRC I was creating a PHAF and trying to find an example of the syntax. To do this I created an annotation in  Canto which is presumably the one I copied (because I  could not find one in the pHAF)

**Comment by ValWood** [comment]
So I know I have used  residue with extensions in Canto. I'm still not sure if these get into the PHAF, because I don't have an example...

**Comment by kimrutherford** [comment]
I can only find one phenotype annotation in Chado that includes a residue extension. It's here:
https://curation.pombase.org/pombe/curs/2fed9a1e695623df/feature/gene/view/2/ro

There might be more in non-approved sessions?

That residue extension isn't in the PHAF file so I'll fix the exporter.

**Comment by ValWood** [comment]
I'm pretty sure I have done a couple recently and I  assumed there were other older ones because the  ability to specify residues was presumably added for specific use cases.

@Antonialock @mah11  can you remember using the residue extension on phenotypes?
When was it implemented?

**Comment by kimrutherford** [comment]
> I'm pretty sure I have done a couple recently and I assumed there were other older ones because the ability to specify residues was presumably added for specific use cases.

Sorry!  My Chado query was dodgy.  There are actually lots.

**Comment by kimrutherford** [comment]
> That residue extension isn't in the PHAF file so I'll fix the exporter.

That's done now.  I'll check the output tomorrow then close the issue.
